src/pydata_sphinx_theme/__init__.py

```diff
@@ -501,6 +502,50 @@ def get_edit_url():
     context["theme_show_toc_level"] = int(context.get("theme_show_toc_level", 1))
 
 
+# -----------------------------------------------------------------------------
+# handle pygment css
+# -----------------------------------------------------------------------------
+def _get_styles(formatter, prefix):
+    """
+    Get styles out of a formatter, where everything has the correct prefix.
+    """
+
+    for line in formatter.get_linenos_style_defs():
+        yield f"{prefix} {line}"
+    yield from formatter.get_background_style_defs(prefix)
+    yield from formatter.get_token_style_defs(prefix)
+
+
+def get_pygments_stylesheet():
+    """
+    Generate the theme-specific pygments.css.
+    There is no way to tell Sphinx how the theme handles modes
+    """
+    light_formatter = HtmlFormatter(style="tango")
+    dark_formatter = HtmlFormatter(style="native")
+
+    lines = []
+
+    light_prefix = 'body[data-theme="light"] .highlight'
+    lines.extend(_get_styles(light_formatter, prefix=light_prefix))
+
+    dark_prefix = 'body[data-theme="dark"] .highlight'
+    lines.extend(_get_styles(dark_formatter, prefix=dark_prefix))
+
+    return "\n".join(lines)
+
+
+# cannot deal with pygments and modes so weoverwrite the pygment css file
```

Comment:
```suggestion
# cannot deal with pygments and modes so we overwrite the pygment css file
```

Reply:
Can you be more descriptive than "cannot deal with"? What specifically doesn't work?